firmware/defmt-test/macros/src/lib.rs

```diff
@@ -200,19 +188,23 @@ fn tests_impl(args: TokenStream, input: TokenStream) -> parse::Result<TokenStrea
             quote!(#ident())
         };
         unit_test_calls.push(quote!(
+            #( #cfgs )*
             #krate::export::check_outcome(#call, #should_error);
         ));
     }
-    let unit_test_names = tests.iter().map(|test| &test.ident);
-    let unit_test_inputs = tests
-        .iter()
-        .map(|test| test.input.as_ref().map(|input| &input.arg));
-    let unit_test_outputs = tests.iter().map(|test| &test.ret_ty);
-    let unit_test_blocks = tests.iter().map(|test| &test.block);
+
+    let test_functions = tests.iter().map(|test| &test.func);
     let unit_test_running = tests
         .iter()
         .enumerate()
-        .map(|(i, test)| format!("({}/{}) running `{}`...", i + 1, tests.len(), test.ident))
+        .map(|(i, test)| {
+            format!(
+                "({}/{}) running `{}`...",
+                i + 1,
+                tests.len(),
```

Comment:
the number of tests that run are not known in this context because cfg are evaluated after expansion

Reply:
Yeah, I changed it so it gets computed at runtime (or in a `const`) in adce0f3